### PR TITLE
SOW-24: adds handling of midway form jumping

### PR DESCRIPTION
**Pull request by osouthwell-scottlogic**
Adds handling of jumping midway into the form
Clears state when you go to the home page

### CODE REVIEW

File: src/context/FormContext.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const FormContext = createContext<{
   timetableEvents: DevelopmentPlanTimetable[];
   loadedDevelopmentPlan: DevelopmentPlan[] | null;
   loadedTimetableEvents: DevelopmentPlanTimetable[] | null;
+  userFlow: string | null;
```
Comment (vferraro-scottlogic) [inline src/context/FormContext.tsx:32]: I think we have an enum we can use as type here?

File: src/pages/FormPageHoc.tsx [hunk 2]
```diff
@@ -25,16 +26,10 @@ export const FormPageHoC = <P extends Record<string, unknown>>(
...
-      loadedDevelopmentPlan,
-      loadedTimetableEvents,
-    } = useFormContext();
+    const { developmentPlan, timetableEvents, userFlow } = useFormContext();
 
-    const isCreateFlow = !loadedDevelopmentPlan && !loadedTimetableEvents;
-
-    const { previousPage, navigateNext } = useSequence(isCreateFlow);
+    //const isCreateFlow = userFlow === Journeys.Create;
+    const { previousPage, navigateNext } = useSequence(Journeys.Create);
```
Comment (Ro4052) [inline src/pages/FormPageHoc.tsx:32]: This is hard-coded to the create journey?

File: src/pages/FormPageHoc.tsx [hunk 3]
```diff
@@ -52,6 +47,14 @@ export const FormPageHoC = <P extends Record<string, unknown>>(
       navigateNext?.();
     };
 
+    const navigate = useNavigate();
+    useEffect(() => {
+      console.log(userFlow);
```
Comment (Ro4052) [inline src/pages/FormPageHoc.tsx:52]: Console log left in?

File: src/pages/FormPageHoc.tsx [hunk 4]
```diff
@@ -52,6 +47,14 @@ export const FormPageHoC = <P extends Record<string, unknown>>(
...
+      if (!userFlow) {
+        navigate(PageRoute.Root);
+      }
+    });
```
Comment (Ro4052) [inline src/pages/FormPageHoc.tsx:56]: Should `useFlow` be a dep of the `useEffect`?

File: src/pages/FormPageHoc.tsx (hunk 3 above)
Comment (vferraro-scottlogic) [inline src/pages/FormPageHoc.tsx:52]: we probably want to remove the logging as well

File: src/pages/create-timetable-page/CreateTimetablePage.tsx [hunk 5]
```diff
@@ -1,9 +1,30 @@
...
+    setLoadedDevelopmentPlan,
+    setUserFlow,
+  } = useFormContext();
+
+  useEffect(() => {
+    setTimetableEvents(DEFAULT_TIMETABLE_EVENTS);
+    setDevelopmentPlan(DEFAULT_DEVELOPMENT_PLAN);
+    setLoadedTimetableEvents(null);
+    setLoadedDevelopmentPlan(null);
+  });
```
Comment (Ro4052) [inline src/pages/create-timetable-page/CreateTimetablePage.tsx:26]: I think we might want empty deps on this so it doesn't happen every render?

Also something to think about in the future is whether we need to do the same thing on the other content pages

File: src/routes/routes.ts [hunk 6]
```diff
@@ -13,3 +13,8 @@ export enum PageRoute {
   UploadTimetable = "/upload-timetable", // TBC
   Export = "/export-your-timetable",
 }
+
+export enum Journeys {
```
Comment (Ro4052) [inline src/routes/routes.ts:17]: To align with the other enum in this file, should we call this `Journey`?

**Review by Ro4052 — APPROVED** [review]

**Review by vferraro-scottlogic — APPROVED** [review]

**Pull request merged**